Formula/rethinkdb.rb

```diff
@@ -20,29 +20,30 @@ class Rethinkdb < Formula
   end
 
   depends_on "boost" => :build
-  depends_on :macos # Due to Python 2 (v8 and gyp fail to build)
-  # https://github.com/Homebrew/linuxbrew-core/pull/19614
-  # https://github.com/rethinkdb/rethinkdb/pull/6401
   depends_on "openssl@1.1"
 
+  uses_from_macos "python" => :build
   uses_from_macos "curl"
-
-  # patch submitted to upstream, https://github.com/rethinkdb/rethinkdb/pull/6934
-  # remove in next release
-  patch do
-    url "https://github.com/chenrui333/rethinkdb/commit/d7d22b4.patch?full_index=1"
-    sha256 "45d387c3360cf6265e27485c273c37224b2ef0000ebd4fdd361dcecc4d2e9791"
-  end
+  uses_from_macos "ncurses"
+  uses_from_macos "zlib"
 
   def install
+    # Can use system Python 2 for older macOS. See https://rethinkdb.com/docs/build
+    ENV["PYTHON"] = which("python3") if !OS.mac? || MacOS.version >= :catalina
+
     # rethinkdb requires that protobuf be linked against libc++
     # but brew's protobuf is sometimes linked against libstdc++
```

Comment:
This comment must be at least 10 years old.

Reply:
The test.proto used to check functionality hasn't been updated and causes configure error:
```
Test protobuf:                  error
* Error: Unable to compile sample protobuf file. Try running ./configure with the --fetch protoc option
[libprotobuf WARNING google/protobuf/compiler/parser.cc:649] No syntax specified for the proto file: mk/gen/protoc/test.proto. Please use 'syntax = "proto2";' or 'syntax = "proto3";' to specify a syntax version. (Defaulted to proto2 syntax.)
In file included from ./mk/gen/protoc//test.pb.cc:4:
In file included from ./mk/gen/protoc/test.pb.h:23:
In file included from /usr/local/include/google/protobuf/io/coded_stream.h:154:
In file included from /usr/local/include/google/protobuf/stubs/common.h:48:
* Aborting configure
```

The version they fetch is actually the same as our current version (3.19.4), but they bypass check in this case.

Reply:
Fair enough, this isn't a blocker.

Though that error doesn't really say anything because it's just a warning, and they limit # of lines output: https://github.com/rethinkdb/rethinkdb/blob/9090794e52078e9ba10bf34e4f6c399f0f71c29e/configure#L416

Reply:
I increased errors and looks like test is failing on default c++ standard:
```
/usr/local/include/google/protobuf/stubs/port.h:123:2: error: "Protobuf requires at least C++11."
```

Could try a run with `ENV.cxx11`.

Reply:
Worth a shot I'd say.